content/algorithms/flow/dinitz_alg.md

```diff
@@ -0,0 +1,331 @@
+---
+jupytext:
+  notebook_metadata_filter: all
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.11.2
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+language_info:
+  codemirror_mode:
+    name: ipython
+    version: 3
+  file_extension: .py
+  mimetype: text/x-python
+  name: python
+  nbconvert_exporter: python
+  pygments_lexer: ipython3
+  version: 3.8.5
+---
+
+# Tutorial: Dinitz's algorithm and its applications
+In this tutorial, we will introduce [Maximum flow problem](https://en.wikipedia.org/wiki/Maximum_flow_problem) 
+and [Dinitz's algorithm](https://en.wikipedia.org/wiki/Dinic%27s_algorithm) [^1], which is implimented at 
+[algorithms/flow/dinitz_alg.py](https://github.com/networkx/networkx/blob/main/networkx/algorithms/flow/dinitz_alg.py) 
+in NetworkX. We will also see how it can be used to solve some interesting problems.
+
+## Maximum flow problem
+
+### Motivation
+Let's say you want to send your friend some data as soon as possible, but the only way 
+of communication/sending data between you two is through a peer-to-peer network. An 
+interesting thing about this peer-to-peer network is that it allows you to send data 
+along the paths you specify with certain limits on the sizes of data per second that 
+you can send between a pair of nodes in this network.
+
+![image with you & friend and network of computer](images/intro.png)
```

Comment:
````suggestion
```{code-cell} ipython3
---
tags: [hide-input]
---
# code to generate the first image goes here!
```
````